shopinvader/models/shopinvader_variant.py

```diff
@@ -293,11 +293,38 @@ def _get_price(
             )
         return res
 
+    @api.model
+    def _get_main_product_read_fields(cls):
+        product_model = cls.env["product.product"]
+        order_by = [x.strip() for x in product_model._order.split(",")]
+        return ["shopinvader_product_id", "backend_id", "lang_id"] + order_by
+
+    @api.model
+    def _get_main_product_sorted_variants(cls, variants):
+        # NOTE: if the order is changed by adding `asc/desc` this can be broken
+        # but it's very unlikely that the default order for product.product
+        # will be changed.
```

Comment:
🤔 that's not a nice assumption

Perhaps it's out of the scope of this PR, but I wonder why this method is even necessary.

[When variants are searched (Line 330-336)](https://github.com/shopinvader/odoo-shopinvader/blob/725a06fe81c0d62799b24c7c256e2bf7983ae6a3/shopinvader/models/shopinvader_variant.py#L330-L336), the order is explicitly set to `order="shopinvader_product_id"`. which should be enough for the ORM to return the records in the right variant order (according to the `product.product` model order)

Am I missing something?

Reply:
No, this is what I copied from the original method.
I don't like this implementation, but we don't have the budget to improve it for now.
Using `sql` syntax order and converting it to a python sort lambda function looks very risky to me…

Reply:
It's stable guys, please don't touch it! Merci! :smile:

Reply:
The reason for the 1st sorting in the search is to be able to use groupby, the 2nd sorting is the real sorting that normally happens when you pick up variants. @mmequignon you can add this my old comment :wink:

Reply:
@simahawk this breaks in version 16 as default order on `product.product` is `_order = 'priority desc, default_code, name, id'`

Reply:
I understand this is a feature distinct from this PR, but if someone wants to improve it, it has been done in v16 by @qgroulard (because the default Odoo ordering on `product.product` changed and a `priority desc` was added).
Feel free to take the inspiration from https://github.com/shopinvader/odoo-shopinvader/blob/0f73c49d0a6c81e5ec2fd85dc8318b3232b3dd5c/shopinvader_product/models/product_product.py#L84